packages/client/src/components/themeMessage/themeMessage.tsx

```diff
@@ -11,19 +11,25 @@ export interface IThemeMessage {
 }
 
 export function ThemeMessage(props: IThemeMessage) {
-  const [countLike, setCountLike] = useState(0);
+  const [hasLike, setHasLike] = useState(false);
 
-  const iconHandler = () => setCountLike((prevState) => prevState > 0 ? --prevState : ++prevState);
-  const amountLikes = props.countLikes ? props.countLikes + countLike : countLike;
+  let countLikes = props.countLikes || 0;
+  const iconHandler = () => setHasLike((prevState) => !prevState);
+
+  if (hasLike) {
+    ++countLikes;
+  } else {
+    countLikes = props.countLikes || 0;
+  }
```

Comment:
[Исправление]
Два раза делаешь одно и то же действие `countLikes = props.countLikes || 0;`. Проверяешь на `undefined`, поэтому лучше использовать `??`
И тут можно в `memo` обернуть переменную, чтобы она пересчитывалась при смене `hasLike`
```suggestion
const countLikes = useMemo(() => {
  return (props.countLikes ?? 0) + hasLike;
}, [hasLike, props.countLikes])
```
Саджест написала руками, могла сделать опечатку

Reply:
Складывать number и boolean законно?))

Reply:
Вполне, я сложила в инспекторе )
А вообще есть [унарный +](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Unary_plus)